amazonia/schema.yaml

```diff
@@ -0,0 +1,309 @@
+#                                        'minsize',
+#                                        'maxsize',
+#                                        'health_check_grace_period',
+#                                        'iam_instance_profile_arn',
+#                                        'sns_topic_arn',
+#                                        'sns_notification_types',
+#                                        'elb_log_bucket',
+#                                        'health_check_type',
+#                                        'dependencies'],
+#                     'database_units': ['unit_title',
+#                                        'db_instance_type',
+#                                        'db_engine',
+#                                        'db_port']
+#                     }
+#
+
+
+unit_title: # A string containing the title to use for autoscaling or database units in your stack
+  type: 'string' # If unit title is not provided here, it must be provided as an embeded key in any autoscaling
+  nullable: True # or database units (if they are not also empty).
+  regex: '^([a-zA-Z0-9]+$)' # matches an alphanumeric title
+  empty: False
+image_id: # A string specifying which ami to use for autoscaling groups by its id
+  type: 'string'
+  empty: False
+  regex: '^([a][m][i][\-][a-zA-Z0-9]+$)' # matches an ami number eg: ami-some1234
+instance_type: # A string containing the instance type to use for autoscaling groups.
+  type: 'string'
+  empty: False
+stack_title: # A string containing the name of the cloud formation stack.
+  type: 'string'
+  empty: False
+  regex: '^([a-zA-Z0-9]+$)' # matches an alphanumeric title
+keypair: # A string containing the Key Pair to use for ssh access to your instances.
+  type: 'string'
+  empty: False
+availability_zones: # A list of strings, specifying each availability zone to use when creating resources.
+  type: 'list'
+  empty: False
+  schema:
+    type: 'string'
+    empty: False
+vpc_cidr: # A cidr block to delegate to your VPC.
+  type: 'string'
+  empty: False # Below regex matches a CIDR pattern eg: 10.0.0.0/16
+  regex: '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$'
+public_cidr: # A dictionary containing the name and cidr block to be considered 'public' by your stack.
+  type: 'dict'
+  empty: False
+  schema:
+    name:
+      type: 'string'
+      empty: False
+      regex: '^([a-zA-Z0-9]+$)' # matches an alphanumeric title
+    cidr:
+      type: 'string'
+      empty: False # Below regex matches a CIDR pattern eg: 10.0.0.0/16
+      regex: '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$'
+jump_image_id: # A string specifying which ami to use for your jump box by its id
+  type: 'string'
+  empty: False
+  regex: '^([a][m][i][\-][a-zA-Z0-9]+$)' # matches an ami number eg: ami-some1234
```

Comment:
Should we use this regex? ^(ami-[a-zA-Z0-9]+$) for ami ids. Its the same but a bit cleaner